kong/plugins/key-auth/handler.lua

```diff
@@ -170,6 +171,12 @@ local function do_authentication(conf)
     return nil, ERR_NO_API_KEY
   end
 
+  -- this request contains invalid symbol in API key, HTTP 401
+  if not key:find("^[%w%-%_]+$") then
```

Comment:
`string.find` cannot be optimized by LuaJIT when doing regular matching, might there be performance issues here?

Reply:
Sorry, I've seen your new comment only after merging.  Do you believe that checking the syntax of the API key with some optimized function is warranted?

Reply:
This line of code is on the hot path, so I think yes.

Reply:
If you also agree I will try to optimize here.

Reply:
In my measurements, the check is taking in the order of 230 nanoseconds.  I'm not confident that putting time into optimizing this is time well spent, in particular if it makes the code less readable.

Reply:
I verified 53 nanoseconds, but I found that the took time increased linearly with the size of the key.